crates/ty_project/src/lib.rs

```diff
@@ -81,8 +82,11 @@ pub struct Project {
     /// We box the metadata here because it's a fairly large type and
     /// reducing the size of `Project` helps reduce the size of the
     /// salsa allocated table for `Project`.
-    #[returns(deref)]
-    pub settings: Box<Settings>,
+    ///
+    /// This field is unintialized temporarily after deserialization,
+    /// until `Project::resolve_settings` is called.
```

Comment:
and the reason for the `OnceLock` is so that we can change the field without bumping the salsa revision?

Reply:
Yes, the settings should not change if the metadata has not changed.